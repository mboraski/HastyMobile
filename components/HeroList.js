// Third Party Imports
import React, { Component } from 'react';
import { ScrollView, StyleSheet } from 'react-native';
import _ from 'lodash';

// Relative Imports
import HeroDetail from './HeroDetail';
import { emY } from '../utils/em';

class HeroList extends Component {
    state = {
        heroes: []
    };

    componentWillMount() {
        this.setState({
            heroes: Array(4).fill().map((e, i) => ({
                id: i,
                name: 'Jessica Morgan',
                type: i % 2 ? 'Instant' : 'Driver',
                delivery_time: 4,
                image: 'https://facebook.github.io/react/img/logo_og.png'
            }))
        });
    }

    renderHeroes() {
        return _.map(this.state.heroes, hero =>
            <HeroDetail
                key={hero.id}
                hero={hero}
            />
        );
    }

    render() {
        return (
            <ScrollView contentContainerStyle={styles.container}>
                {this.renderHeroes()}
            </ScrollView>
        );
    }
}

const styles = StyleSheet.create({
    container: {
    }
});

export default HeroList;
