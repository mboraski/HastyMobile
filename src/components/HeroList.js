// Third Party Imports
import React, { Component } from 'react';
import { View, StyleSheet } from 'react-native';
import { connect } from 'react-redux';

// Relative Imports
import HeroDetail from './HeroDetail';

class HeroList extends Component {
    // heroes: [
    //     {
    //         id: 1,
    //         name: 'Jessica Morgan',
    //         type: 'Instant',
    //         delivery_time: 4,
    //         image: 'https://facebook.github.io/react/img/logo_og.png'
    //     },
    //     {
    //         id: 2,
    //         name: 'Matt Littlefield',
    //         type: 'Fast',
    //         delivery_time: 16,
    //         image: 'https://facebook.github.io/react/img/logo_og.png'
    //     }
    // ],

    render() {
        const { hero } = this.props;
        let result;
        console.log('hero: ', hero);
        if (hero) {
            result = (
                <View style={styles.container}>
                    <HeroDetail
                        hero={hero}
                    />
                </View>
            );
        } else {
            result = null;
        }
        return result;
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff'
    }
});

const mapStateToProps = state => ({
    hero: state.order.hero
});

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(HeroList);
