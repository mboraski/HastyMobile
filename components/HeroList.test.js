import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import HeroList from './HeroList';

const initialState = {};

describe('HeroList', () => {
    const middlewares = [];
    const mockStore = configureStore(middlewares);
    it('renders correctly', () => {
        const wrapper = shallow(<HeroList />, {
            context: { store: mockStore(initialState) }
        });
        const render = wrapper.dive();
        expect(render).toMatchSnapshot();
    });
});
