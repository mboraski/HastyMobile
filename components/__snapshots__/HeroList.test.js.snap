// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroList renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Unknown
    contentContainerStyle={Object {}}
>
    <View>
        <HeroDetail
            hero={
                Object {
                    "delivery_time": 4,
                    "id": 0,
                    "image": "https://facebook.github.io/react/img/logo_og.png",
                    "name": "Jessica Morgan",
                    "type": "Driver",
                  }
            }
        />
        <HeroDetail
            hero={
                Object {
                    "delivery_time": 4,
                    "id": 1,
                    "image": "https://facebook.github.io/react/img/logo_og.png",
                    "name": "Jessica Morgan",
                    "type": "Instant",
                  }
            }
        />
        <HeroDetail
            hero={
                Object {
                    "delivery_time": 4,
                    "id": 2,
                    "image": "https://facebook.github.io/react/img/logo_og.png",
                    "name": "Jessica Morgan",
                    "type": "Driver",
                  }
            }
        />
        <HeroDetail
            hero={
                Object {
                    "delivery_time": 4,
                    "id": 3,
                    "image": "https://facebook.github.io/react/img/logo_og.png",
                    "name": "Jessica Morgan",
                    "type": "Instant",
                  }
            }
        />
    </View>
</Unknown>,
  "nodes": Array [
    <Unknown
      contentContainerStyle={Object {}}
>
      <View>
            <HeroDetail
                  hero={
                        Object {
                              "delivery_time": 4,
                              "id": 0,
                              "image": "https://facebook.github.io/react/img/logo_og.png",
                              "name": "Jessica Morgan",
                              "type": "Driver",
                            }
                  }
            />
            <HeroDetail
                  hero={
                        Object {
                              "delivery_time": 4,
                              "id": 1,
                              "image": "https://facebook.github.io/react/img/logo_og.png",
                              "name": "Jessica Morgan",
                              "type": "Instant",
                            }
                  }
            />
            <HeroDetail
                  hero={
                        Object {
                              "delivery_time": 4,
                              "id": 2,
                              "image": "https://facebook.github.io/react/img/logo_og.png",
                              "name": "Jessica Morgan",
                              "type": "Driver",
                            }
                  }
            />
            <HeroDetail
                  hero={
                        Object {
                              "delivery_time": 4,
                              "id": 3,
                              "image": "https://facebook.github.io/react/img/logo_og.png",
                              "name": "Jessica Morgan",
                              "type": "Instant",
                            }
                  }
            />
      </View>
</Unknown>,
  ],
  "options": Object {
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      },
      "_currentElement": <ScrollViewMock
        contentContainerStyle={Object {}}
>
        <HeroDetail
                hero={
                        Object {
                                "delivery_time": 4,
                                "id": 0,
                                "image": "https://facebook.github.io/react/img/logo_og.png",
                                "name": "Jessica Morgan",
                                "type": "Driver",
                              }
                }
        />
        <HeroDetail
                hero={
                        Object {
                                "delivery_time": 4,
                                "id": 1,
                                "image": "https://facebook.github.io/react/img/logo_og.png",
                                "name": "Jessica Morgan",
                                "type": "Instant",
                              }
                }
        />
        <HeroDetail
                hero={
                        Object {
                                "delivery_time": 4,
                                "id": 2,
                                "image": "https://facebook.github.io/react/img/logo_og.png",
                                "name": "Jessica Morgan",
                                "type": "Driver",
                              }
                }
        />
        <HeroDetail
                hero={
                        Object {
                                "delivery_time": 4,
                                "id": 3,
                                "image": "https://facebook.github.io/react/img/logo_og.png",
                                "name": "Jessica Morgan",
                                "type": "Instant",
                              }
                }
        />
</ScrollViewMock>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ScrollViewMock {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": Array [
            <HeroDetail
              hero={
                            Object {
                                          "delivery_time": 4,
                                          "id": 0,
                                          "image": "https://facebook.github.io/react/img/logo_og.png",
                                          "name": "Jessica Morgan",
                                          "type": "Driver",
                                        }
              }
/>,
            <HeroDetail
              hero={
                            Object {
                                          "delivery_time": 4,
                                          "id": 1,
                                          "image": "https://facebook.github.io/react/img/logo_og.png",
                                          "name": "Jessica Morgan",
                                          "type": "Instant",
                                        }
              }
/>,
            <HeroDetail
              hero={
                            Object {
                                          "delivery_time": 4,
                                          "id": 2,
                                          "image": "https://facebook.github.io/react/img/logo_og.png",
                                          "name": "Jessica Morgan",
                                          "type": "Driver",
                                        }
              }
/>,
            <HeroDetail
              hero={
                            Object {
                                          "delivery_time": 4,
                                          "id": 3,
                                          "image": "https://facebook.github.io/react/img/logo_og.png",
                                          "name": "Jessica Morgan",
                                          "type": "Instant",
                                        }
              }
/>,
          ],
          "contentContainerStyle": Object {},
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Unknown
          contentContainerStyle={Object {}}
>
          <View>
                    <HeroDetail
                              hero={
                                        Object {
                                                  "delivery_time": 4,
                                                  "id": 0,
                                                  "image": "https://facebook.github.io/react/img/logo_og.png",
                                                  "name": "Jessica Morgan",
                                                  "type": "Driver",
                                                }
                              }
                    />
                    <HeroDetail
                              hero={
                                        Object {
                                                  "delivery_time": 4,
                                                  "id": 1,
                                                  "image": "https://facebook.github.io/react/img/logo_og.png",
                                                  "name": "Jessica Morgan",
                                                  "type": "Instant",
                                                }
                              }
                    />
                    <HeroDetail
                              hero={
                                        Object {
                                                  "delivery_time": 4,
                                                  "id": 2,
                                                  "image": "https://facebook.github.io/react/img/logo_og.png",
                                                  "name": "Jessica Morgan",
                                                  "type": "Driver",
                                                }
                              }
                    />
                    <HeroDetail
                              hero={
                                        Object {
                                                  "delivery_time": 4,
                                                  "id": 3,
                                                  "image": "https://facebook.github.io/react/img/logo_og.png",
                                                  "name": "Jessica Morgan",
                                                  "type": "Instant",
                                                }
                              }
                    />
          </View>
</Unknown>,
        "_debugID": 4,
        "_renderedOutput": <Unknown
          contentContainerStyle={Object {}}
>
          <View>
                    <HeroDetail
                              hero={
                                        Object {
                                                  "delivery_time": 4,
                                                  "id": 0,
                                                  "image": "https://facebook.github.io/react/img/logo_og.png",
                                                  "name": "Jessica Morgan",
                                                  "type": "Driver",
                                                }
                              }
                    />
                    <HeroDetail
                              hero={
                                        Object {
                                                  "delivery_time": 4,
                                                  "id": 1,
                                                  "image": "https://facebook.github.io/react/img/logo_og.png",
                                                  "name": "Jessica Morgan",
                                                  "type": "Instant",
                                                }
                              }
                    />
                    <HeroDetail
                              hero={
                                        Object {
                                                  "delivery_time": 4,
                                                  "id": 2,
                                                  "image": "https://facebook.github.io/react/img/logo_og.png",
                                                  "name": "Jessica Morgan",
                                                  "type": "Driver",
                                                }
                              }
                    />
                    <HeroDetail
                              hero={
                                        Object {
                                                  "delivery_time": 4,
                                                  "id": 3,
                                                  "image": "https://facebook.github.io/react/img/logo_og.png",
                                                  "name": "Jessica Morgan",
                                                  "type": "Instant",
                                                }
                              }
                    />
          </View>
</Unknown>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ScrollViewMock
    contentContainerStyle={Object {}}
>
    <HeroDetail
        hero={
            Object {
                "delivery_time": 4,
                "id": 0,
                "image": "https://facebook.github.io/react/img/logo_og.png",
                "name": "Jessica Morgan",
                "type": "Driver",
              }
        }
    />
    <HeroDetail
        hero={
            Object {
                "delivery_time": 4,
                "id": 1,
                "image": "https://facebook.github.io/react/img/logo_og.png",
                "name": "Jessica Morgan",
                "type": "Instant",
              }
        }
    />
    <HeroDetail
        hero={
            Object {
                "delivery_time": 4,
                "id": 2,
                "image": "https://facebook.github.io/react/img/logo_og.png",
                "name": "Jessica Morgan",
                "type": "Driver",
              }
        }
    />
    <HeroDetail
        hero={
            Object {
                "delivery_time": 4,
                "id": 3,
                "image": "https://facebook.github.io/react/img/logo_og.png",
                "name": "Jessica Morgan",
                "type": "Instant",
              }
        }
    />
</ScrollViewMock>,
}
`;
